pkg/pf/tests/diff_test/diff_test.go

```diff
@@ -59,6 +68,275 @@ Resources:
+				Optional:    true,
+				NestedObject: rschema.NestedAttributeObject{
+					Attributes: map[string]rschema.Attribute{
+						"key": rschema.StringAttribute{
+							Description: "Key of the custom header.",
+							Required:    true,
+						},
+						"value": rschema.StringAttribute{
+							Required:      true,
+							Description:   "Value of the custom header. This value cannot be imported.",
+							Sensitive:     true,
+							PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+						},
+					},
+				},
+			},
+		}
+		if withNew {
+			attributes["resource_access_token_events"] = rschema.BoolAttribute{
+				Default:  booldefault.StaticBool(false),
+				Computed: true,
+				Optional: true,
+			}
+		}
+		return rschema.Schema{
+			Attributes: attributes,
+		}
+	}
+
+	res := pb.NewResource(pb.NewResourceArgs{
+		CreateFunc: func(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+			resp.State = tfsdk.State(req.Config)
+			resp.State.SetAttribute(ctx, path.Root("project_id"), 123)
+			resp.State.SetAttribute(ctx, path.Root("hook_id"), 567)
+			resp.State.SetAttribute(ctx, path.Root("id"), "abc")
+			resp.State.SetAttribute(ctx, path.Root("enable_ssl_verification"), true)
+			resp.State.SetAttribute(ctx, path.Root("confidential_note_events"), false)
+			resp.State.SetAttribute(ctx, path.Root("releases_events"), false)
+			resp.State.SetAttribute(ctx, path.Root("deployment_events"), false)
+		},
+		UpdateFunc: func(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+			resp.State = tfsdk.State(req.Plan)
+		},
+		ResourceSchema: getSchema(false),
+	})
+
+	inputs := map[string]cty.Value{
+		"project":                    cty.StringVal("project_id"),
+		"name":                       cty.StringVal("webhook-receiver"),
+		"confidential_issues_events": cty.BoolVal(true),
+		"issues_events":              cty.BoolVal(true),
+		"job_events":                 cty.BoolVal(true),
+		"merge_requests_events":      cty.BoolVal(true),
+		"note_events":                cty.BoolVal(true),
+		"pipeline_events":            cty.BoolVal(true),
+		"push_events":                cty.BoolVal(true),
+		"tag_push_events":            cty.BoolVal(true),
+		"url":                        cty.StringVal("https://webhook.receiver.endpoint/hooks/gitlab"),
+		"wiki_page_events":           cty.BoolVal(true),
+	}
+	diff := crosstests.Diff(t, res,
+		inputs,
+		inputs,
+		crosstests.DiffUpdateResource(pb.NewResource(pb.NewResourceArgs{
+			ResourceSchema: getSchema(true),
+		})),
+		crosstests.DiffSkipUp(true),
+	)
+
+	autogold.Expect(`
+Terraform used the selected providers to generate the following execution
+plan. Resource actions are indicated with the following symbols:
++/- create replacement and then destroy
+
+Terraform will perform the following actions:
+
+  # testprovider_test.res must be replaced
++/- resource "testprovider_test" "res" {
```

Comment:
The cross tests show the Terraform diff with a replace plan, same as pulumi. Why would this be different than what we get using a real Terraform program?